src/lib.rs

```diff
@@ -220,10 +223,35 @@ cfg_if! {
     if #[cfg(any(target_os = "haiku", target_os = "redox", target_os = "nto", target_os = "aix"))] {
         mod util_libc;
         #[path = "use_file.rs"] mod imp;
-    } else if #[cfg(any(target_os = "android", target_os = "linux"))] {
+    } else if #[cfg(all(
+        not(feature = "disable_urandom_fallback"),
+        any(
+            // Rust supports Android API level 19 (KitKat) [0], while `getrandom(2)`
+            // was added only in level 23 (Marshmallow).
+            // [0]: https://blog.rust-lang.org/2023/01/09/android-ndk-update-r25.html
```

Comment:
I also see that rust-lang/rust is about to bump up to API level 21 in https://github.com/rust-lang/rust/pull/120593, which is still not enough.

Reply:
Added link to the Rust PR.